packages/core/src/components/forms/numericInput.tsx

```diff
@@ -374,8 +418,13 @@ export class NumericInput extends AbstractPureComponent<HTMLInputProps & INumeri
 
     private handleButtonKeyUp = (
         e: React.KeyboardEvent<HTMLInputElement>,
+        direction: IncrementDirection,
         onClick: React.MouseEventHandler<HTMLInputElement>,
     ) => {
+        this.updateDelta(direction, e);
+        // respond explicitly on key *up*, because onKeyDown triggers multiple
+        // times and doesn't always receive modifier-key flags, leading to an
+        // unintuitive/out-of-control incrementing experience.
```

Comment:
this comment does not make sense to me. what is this out-of-control experience? 

would we be hurt down the line if this comment were removed?